saleor/graphql/checkout/mutations.py

```diff
@@ -193,31 +215,46 @@ def clean_input(cls, info, instance, data):
         return cleaned_input
 
     @classmethod
-    @transaction.atomic()
-    def save(cls, info, instance, cleaned_input):
+    def save_addresses(cls, instance: models.Checkout, cleaned_input: dict):
         shipping_address = cleaned_input.get("shipping_address")
         billing_address = cleaned_input.get("billing_address")
 
-        if shipping_address:
+        updated_fields = []
+
+        if shipping_address and instance.is_shipping_required():
             shipping_address.save()
             instance.shipping_address = shipping_address.get_copy()
+            updated_fields.append("shipping_address")
         if billing_address:
             billing_address.save()
             instance.billing_address = billing_address.get_copy()
+            updated_fields.append("billing_address")
+
+        # Save the updated fields
```

Comment:
Do we need these comments? I think that code is enough readable and we don't need additional comments..

Reply:
I added those, in case someone goes through this and is wondering about the case where `updated_fields` is empty.

I could remove `Save [...]` for sure